crates/common/tedge_config_macros/impl/src/query.rs

```diff
@@ -292,42 +351,72 @@ fn keys_enum(
     }
 }
 
+#[derive(Debug, Default)]
+struct NumericIdGenerator {
+    count: u32,
+}
+
+pub trait IdGenerator: Default {
+    fn next_id(&mut self, span: Span) -> syn::Ident;
+}
+
+impl IdGenerator for NumericIdGenerator {
```

Comment:
I will likely adopt this approach more widely, since the PR currently has a bunch of ways of assigning these in various places.

I should also document what this is doing.